api/nntrainer-api-common.h

```diff
@@ -151,6 +151,19 @@ typedef enum {
 typedef int (*ml_train_datagen_cb)(float **input, float **label, bool *last,
                                    void *user_data);
 
+/**
+ * @brief Enumeration for the dataset data type of NNTrainer.
+ * @since_tizen 6.5
+ */
+typedef enum {
+  ML_TRAIN_DATASET_DATA_USAGE_TRAIN =
```

Comment:
ML_TRAIN_DATASET_TYPE_TRAIN?

Reply:
Would there be alternatives? I think dataset_type will be used later as 'file, image_directory' kind of stuff

Reply:
Isn't that data type? USAGE just sounds strange but works for now.

Reply:
Please throw me some word :p

`ml_train_dataset_add_generator(dataset, train /**<-usage */, cb, user_data)` will add generator to be used for training.

Reply:
ML_TRAIN_DATASET_FOR_TRAINING.. it sounds strange.. ML_**TRAIN**_DATASET_DATA_FOR_**TRAIN**...
VALIDATING.. TESTING.. Maybe?

Reply:
I think it good idae!
As there are number of changes will break in the upcoming PR, I'll do this before making acr :)

Reply:
As per offline discussion, this will be
```
typedef enum {
  ML_TRAIN_DATASET_MODE_TRAIN =
    0, /**< The given data is for used when training */
  ML_TRAIN_DATASET_MODE_VALID =
    1, /**< The given data is for used when validating */
  ML_TRAIN_DATASET_MODE_TEST =
    2, /**< The given data is for used when testing */
} ml_train_dataset_mode_e;
```